packages/nuxt/src/components/runtime/server-component.ts

```diff
@@ -1,102 +1,15 @@
-import { defineComponent, createStaticVNode, computed, h, watch } from 'vue'
-import { debounce } from 'perfect-debounce'
-import { hash } from 'ohash'
-import { appendHeader } from 'h3'
-
-import { useHead } from '@unhead/vue'
-import type { NuxtIslandResponse } from '../../core/runtime/nitro/renderer'
-import { useNuxtApp } from '#app/nuxt'
-import { useRequestEvent } from '#app/composables/ssr'
-import { useAsyncData } from '#app/composables/asyncData'
-
-const pKey = '_islandPromises'
+import { defineComponent, h } from 'vue'
+import NuxtIsland from '#app/components/nuxt-island'
 
 export const createServerComponent = (name: string) => {
   return defineComponent({
     name,
     inheritAttrs: false,
     setup (_props, { attrs }) {
-      return () => h(NuxtServerComponent, {
+      return () => h(NuxtIsland, {
         name,
         props: attrs
       })
     }
   })
 }
-
-const NuxtServerComponent = defineComponent({
-  name: 'NuxtServerComponent',
-  props: {
-    name: {
-      type: String,
-      required: true
-    },
-    props: {
-      type: Object,
-      default: () => undefined
-    },
-    context: {
-      type: Object,
-      default: () => ({})
-    }
-  },
-  async setup (props) {
-    const nuxtApp = useNuxtApp()
-    const hashId = computed(() => hash([props.name, props.props, props.context]))
-
-    const event = useRequestEvent()
-
-    function _fetchComponent () {
-      const url = `/__nuxt_island/${props.name}:${hashId.value}`
-      if (process.server && process.env.prerender) {
-        // Hint to Nitro to prerender the island component
-        appendHeader(event, 'x-nitro-prerender', url)
-      }
-      // TODO: Validate response
-      return $fetch<NuxtIslandResponse>(url, {
-        params: {
-          ...props.context,
-          props: props.props ? JSON.stringify(props.props) : undefined
-        }
-      })
-    }
-
-    const res = useAsyncData(
-      `${props.name}:${hashId.value}`,
-      async () => {
-        nuxtApp[pKey] = nuxtApp[pKey] || {}
-        if (!nuxtApp[pKey][hashId.value]) {
-          nuxtApp[pKey][hashId.value] = _fetchComponent().finally(() => {
-            delete nuxtApp[pKey]![hashId.value]
-          })
-        }
```

Comment:
The main reason there was a separate `NuxtServerComponent` component is to allow us to include its content in the payload so we can prefetch it. This isn't ideal as it inflates the payload, but maybe we could instead add the uid into the payload and also add a plugin to prefetch any prerendered server components?

Reply:
I don't really get why do we need to prefetch it since the HTML part wil be already loaded 🤔 . Did i miss something about it ?

Reply:
I mean, for subsequent navigation. We ideally don't want to click a link and _then_ download html for a server component on another page. Just as we have prefetched the payload in advance, we can prefetch used server components in advance.

Reply:
i see i'll set back NuxtServerComponent with the fix